### PR TITLE
Ability to use server_name instead of server_id

### PR DESCRIPTION
**Pull request by suikast42**
See https://github.com/nats-io/prometheus-nats-exporter/issues/196

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -161,6 +162,13 @@ necessary.`)
 		if err := exp.AddServer(id, url); err != nil {
 			collector.Fatalf("Unable to setup server in exporter: %s, %s: %v", id, url, err)
 		}
+	} else if len(args) == 1 && opts.UseInternalServerID {
```
Comment (wallyqs) [inline main.go:165]: I think you mean the use server name option here?

File: exporter/exporter.go [hunk 2]
```diff
@@ -41,29 +41,30 @@ const (
...
+	CertFile              string
+	KeyFile               string
+	CaFile                string
+	NATSServerURL         string
+	NATSServerTag         string
+	HTTPUser              string // User in metrics scrape by prometheus.
+	HTTPPassword          string
+	Prefix                string
+	UseInternalServerID   bool
+	UseInternalServerName bool
```
Comment (wallyqs) [inline exporter/exporter.go:67]: Could just be:
```suggestion
	UseServerName bool
```

**Comment by wallyqs** [comment]
Thanks for the contribution! Left a couple of comments.

**Comment by suikast42** [comment]
Gladly. This my first pull request to an opensource project. Does that mean that I have to make the these suggested changes and push a new pull request or how to proceed, Thanks.

**Comment by Doslin** [comment]
Can this function be merged into the master branch?

**Comment by mcosta74** [comment]
> Gladly. This my first pull request to an opensource project. Does that mean that I have to make the these suggested changes and push a new pull request or how to proceed, Thanks.

@suikast42  Usually you have just to make the requested changes and then push on the branch; then, depending on the maintainers' policy, the commits will be squashed before to go in produciton

**Comment by mcosta74** [comment]
Would be great to have it merged, may I help in some way ? @suikast42 @wallyqs ?

**Comment by codegangsta** [comment]
@suikast42 Thanks for the contribution. I you don't mind. I'm happy to make the changes based on @wallyqs's suggestions and get this merged

**Comment by suikast42** [comment]
ist there schedule date for the next version with hat  merge ?

**Comment by mcosta74** [comment]
> is there schedule date for the next version with hat merge ?

the PR is not merged yet because there're changes requested by @wallyqs . do you need help to address them ?

**Comment by suikast42** [comment]
> > is there schedule date for the next version with hat merge ?
> 
> the PR is not merged yet because there're changes requested by @wallyqs . do you need help to address them ?

Sure

**Comment by mcosta74** [comment]
@suikast42 / @codegangsta  should I apply the canges or some of you is doing it ?

**Comment by suikast42** [comment]
> @suikast42 / @codegangsta should I apply the changes or some of you is doing it ?

I appriciate it

**Comment by codegangsta** [comment]
> @suikast42 / @codegangsta  should I apply the changes or some of you is doing it ?

I haven't started any work yet if you want to. But I can get to it this week

**Comment by mcosta74** [comment]
I'll work on it

**Comment by mcosta74** [comment]
I can't push on the PR's branch so I've created a new PR #211.

I fixed some linter violations and added a Unit Test

**Comment by wallyqs** [comment]
Fixed via #211

**Pull request closed without merging**